widgets/contact/contact.php

```diff
@@ -1227,6 +1227,12 @@ function send_mail( $email_fields, $instance ) {
 			$instance['settings']['from'] = get_option( 'admin_email' );
 		}
 
+		if ( $instance['settings']['from'] == $instance['settings']['from'] ) {
+			$domain = parse_url( esc_url( get_site_url() ), PHP_URL_HOST );
+			$domain = str_replace( 'www.', '', $domain );
+			$instance['settings']['from'] = "wordpress@$domain";
```

Comment:
We shouldn't do this here. We should rather check this when we're setting up the default values of the form, so that users have the opportunity to see and change it before any emails are sent.

Reply:
To clarify, are you suggesting that form validation should be implemented or after form submission?

Reply:
Hehehe, nope. Form validation would be nice, but that's a pretty big addition for another day. 😜 

I mean when the form is rendered we populate some fields, e.g. the from and to fields, with defaults. At that point we should be checking whether `from` is the same `to` and if it is, set the `from` field to wordpress@domain.com.

Reply:
So just after this:
https://github.com/siteorigin/so-widgets-bundle/blob/af811014ff18290d304f9d9eb36f4f4359251e0c/widgets/contact/contact.php#L686-L692

Reply:
Ah. Should the following be moved also?

https://github.com/siteorigin/so-widgets-bundle/blob/af811014ff18290d304f9d9eb36f4f4359251e0c/widgets/contact/contact.php#L1226-L1234

Reply:
Hmm... yeah, but it seems we should have some level of redundancy between here and the `modify_instance` function, where the defaults are set. So perhaps extract all the related email field value sanitization we're doing into a function, which will then be called both here and in `modify_instance`.